### PR TITLE
JWT Cookie Authentication

### LINKED ISSUE #55: Added Per-cookie authentication
As mentioned in this article (https://dev.to/rdegges/please-stop-using-local-storage-1i04), storing sensitive data in localstorage is a bad idea.
I think it would be good to allow the JWT token to be stored in a cookie with the "httpOnly" and "secure" flags to counter XSS attacks.

### PR DESCRIPTION
**Pull request by mongkok**
Resolve #55

When a token is requested and `jwt_cookie` decorator is set, the response will set the given cookie with the token string:

```py
from django.urls import path

from graphene_django.views import GraphQLView
from graphql_jwt.decoratos import jwt_cookie

urlpatterns = [
    path('graphql/', jwt_cookie(GraphQLView.as_view())),
]
```

Update: Used `graphene_django.views` instead of `graphql_extensions.views`.

### CODE REVIEW

File: graphql_jwt/decorators.py [hunk 1]
```diff
@@ -62,14 +64,14 @@ def check_perms(user):
 
 def token_auth(f):
     @wraps(f)
+    @setup_jwt_cookie
     def wrapper(cls, root, info, password, **kwargs):
         def on_resolve(values):
             user, payload = values
             payload.token = get_token(user, info.context)
```
Comment (mtszsobczak) [inline graphql_jwt/decorators.py]: Don't we need to get the token from cookie in get_token?

File: docs/cookie_authentication.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Cookie authentication
```
Comment (mtszsobczak) [inline docs/cookie_authentication.rst]: I am really glad that you are working on this! Thanks :). I did some testing and it seems that when token is only in cookie request.user is an AnonymousUser. Cookie can be seen on request.cookies though. I am not very familiar with the code base so I am not sure whether the middleware or just get_token need extension (or none :) ). I am happy to help with this one.

File: graphql_jwt/decorators.py (hunk 1 above)
Comment (mongkok) [reply graphql_jwt/decorators.py]: Not here, this is to request a new token.

File: docs/cookie_authentication.rst (hunk 2 above)
Comment (mongkok) [reply docs/cookie_authentication.rst]: Thanks @mtszsobczak !,
I forgot to read the token from the cookie 😅, I'll update the PR soon.

File: graphql_jwt/decorators.py (hunk 1 above)
Comment (mtszsobczak) [reply graphql_jwt/decorators.py]: I was talking about adding the logic inside get_token function. Though it seems like not the right place. I did it like that: https://github.com/mtszsobczak/django-graphql-jwt/commit/928fc4c3ccc0378a00fb077c3b0443fc6e3d099b looks like it is working ;)

File: docs/cookie_authentication.rst (hunk 2 above)
Comment (mtszsobczak) [reply docs/cookie_authentication.rst]: Np :) happy to help out

File: graphql_jwt/decorators.py [hunk 3]
```diff
@@ -96,3 +98,33 @@ def on_resolve(values):
...
+
+
+def jwt_cookie(view_func):
+    @wraps(view_func)
+    def wrapped_view(request, *args, **kwargs):
+        request.jwt_cookie = True
+        response = view_func(request, *args, **kwargs)
+
+        if hasattr(request, 'jwt'):
+            expiration = datetime.utcnow() + jwt_settings.JWT_EXPIRATION_DELTA
```
Comment (mtszsobczak) [inline graphql_jwt/decorators.py]: :+1:

File: docs/cookie_authentication.rst [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+=====================
+
+When a token is requested and ``jwt_cookie`` decorator is set, the response will set the given cookie with the token string::
```
Comment (mtszsobczak) [inline docs/cookie_authentication.rst]: What do you think about mentioning here (or somewhere else) that CSRF token should be used when using cookie as transport mechanism for jwt token?

File: graphql_jwt/decorators.py (hunk 1 above)
Comment (mongkok) [reply graphql_jwt/decorators.py]: Yeah, that should work :)

File: graphql_jwt/decorators.py [hunk 5]
```diff
@@ -97,3 +98,33 @@ def on_resolve(values):
...
+
+            response.set_cookie(
```
Comment (mtszsobczak) [inline graphql_jwt/decorators.py]: What do you think about making the cookie secure only as an opt-in? Similar to https://docs.djangoproject.com/en/2.1/ref/settings/#std:setting-SESSION_COOKIE_SECURE

**Comment by mtszsobczak** [comment]
Tested and it seems to be working. Great work!

File: graphql_jwt/decorators.py (hunk 5 above)
Comment (mongkok) [reply graphql_jwt/decorators.py]: Good idea, thanks!

**Pull request merged**